### PR TITLE
ENT-9167: Added ci scripts and github workflow to use them

### PR DESCRIPTION
**Pull request by craigcomstock**
With the idea that this can be re-used by other projects as well as locally for developers.

Ticket: ENT-9617
Changelog: none

merge with https://github.com/cfengine/mission-portal/pull/1996

### CODE REVIEW

**Comment by craigcomstock** [comment]
@Lex-2008 it seems that I can't share a secret from another repo's workflow calling this one when this one is a fork maybe? https://github.com/craigcomstock/mission-portal/actions/runs/5405047467/jobs/9821221095

**Review by Lex-2008 — APPROVED** [review]
nice stuff!

File: ci/build.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+# TODO maybe seed the cache? cp -R buildscripts/ci/cache ~/.cache
+time ./buildscripts/build-scripts/install-dependencies
+time ./buildscripts/build-scripts/configure # 3 minutes locally
+time ./buildscripts/build-scripts/generate-source-tarballs # 1m49
+time ./buildscripts/build-scripts/compile
+time sudo apt remove -y 'cfbuild*' || true
+time sudo apt remove -y 'cfengine-*' || true
+time sudo rm -rf /var/cfengine
+time sudo rm -rf /opt/cfengine
+time ./buildscripts/build-scripts/install-dependencies
```
[the remark this answers is not in the document]
Comment (craigcomstock) [reply ci/build.sh:19]: It's more of an iterative development step. If you happen to install a cfengine-nova package or break in the middle of a build and have a few cfbuild- packages or something. In jenkins I think we are always pretty darn clean.

File: ci/setup.sh [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# setup build host on ubuntu 20
+set -ex
+PREFIX=/var/cfengine
+
+# Github Actions provides machines with various packages installed,
+# what confuses our build system into thinking that it's an RPM distro.
+sudo rm -f /bin/rpm
+export EXPLICIT_ROLE=hub
```
[the remark this answers is not in the document]
Comment (craigcomstock) [reply ci/setup.sh:8]: because that is the default? Ah, no, because we aren't doing anything with that env var! :O Ack.

**Comment by craigcomstock** [comment]
Success! mission-portal PR works with these changes: https://github.com/cfengine/mission-portal/actions/runs/5589562312/jobs/10250286606 🟢

**Comment by craigcomstock** [comment]
let's prefer the in-repo PR: https://github.com/cfengine/buildscripts/pull/1239 since that will tell us if the workflow works in buildscripts by itself.

**Pull request closed without merging**